### ISSUE TITLE
non-nullable camelized field names (in mutations) result in InvalidNullError

### ISSUE THREAD

**Issue by robinboening** [rmosolgo/graphql-ruby#3165]
**Describe the bug**

Probably closely related #2995

```
"message": "uninitialized constant Mutations:: ActivateProvider::InvalidNullError",
"backtrace": [
  "/usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:257:in `continue_value'",
  "/usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:231:in `block (4 levels) in evaluate_selections'",
  "/usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:458:in `after_lazy'",
  "/usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:230:in `block (3 levels) in evaluate_selections'",
  "/usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:383:in `resolve_with_directives'",
  ...
]
```

**Versions**

`graphql (1.10.14)`
`rails (6.0.3.2)`

**GraphQL schema**

```ruby
module Mutations
  class ActivateProvider < BaseMutation # (GraphQL::Schema::RelayClassicMutation)
    description "..."

    argument :provider, String, "The provider name", required: true

    field :redirect_url, String, null: false

    def resolve(provider:)
        {
          redirectUrl: "foo"
        }
    end
end
```

**GraphQL query**

Example GraphQL query and response (if query execution is involved)

```graphql
mutation activateProvider(
  $provider_input: ActivateProviderInput!
) {
  activateProvider(input:$provider_input) {
    redirectUrl
    errors
  }
}

# variables
{
  "provider_input": {
    "provider": "Xyz"
  }
}
```

**Steps to reproduce**

Run the above query and you should see the below mentioned `Mutations::ActivateProvider::InvalidNullError` error.
Its worth to mention this error only happens when using a camelized field! Using `url` instead of `redirectUrl` works just fine.

**Expected behavior**

I expect to see

```json
{
  "data": {
    "activateProvider": {
      "redirectUrl": "foo"
    }
  }
}
```

**Actual behavior**

What specifically went wrong?

<details>
<summary>Click to view exception backtrace</summary>

```
uninitialized constant Mutations::ActivateProvider::InvalidNullError
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:257:in `continue_value'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:231:in `block (4 levels) in evaluate_selections'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:458:in `after_lazy'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:230:in `block (3 levels) in evaluate_selections'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:383:in `resolve_with_directives'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:219:in `block (2 levels) in evaluate_selections'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:458:in `after_lazy'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:177:in `block in evaluate_selections'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:124:in `each'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:124:in `evaluate_selections'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:335:in `block in continue_field'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:458:in `after_lazy'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:330:in `continue_field'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:236:in `block (4 levels) in evaluate_selections'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:458:in `after_lazy'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:230:in `block (3 levels) in evaluate_selections'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:383:in `resolve_with_directives'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:219:in `block (2 levels) in evaluate_selections'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:458:in `after_lazy'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:177:in `block in evaluate_selections'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:124:in `each'
foobar-api_1  | /usr/local/bundle/gems/graphql-1.10.14/lib/graphql/execution/interpreter/runtime.rb:124:in `evaluate_selections'
```

</details>

Thanks for checking! ❤️

**Comment by rmosolgo** [comment]
Thanks for the detailed report! I have two suggestions: 

- Fix the constant missing error by upgrading to 1.11.1 (It was fixed in #2997)
- Return `{ redirect_url: ... }` from the mutation (with `redirect_url` instead of `redirectUrl`) so that GraphQL can correctly find the returned value 

If you try one or both of those and they don't work as expected, let me know!

**Comment by robinboening** [comment]
ohh, I thought I was on the latest release. I'll try and report back. Thanks for the fast response! :)
